pkg/runtime/pubsub/cloudevents.go

```diff
@@ -14,28 +14,36 @@ limitations under the License.
 package pubsub
 
 import (
-	"github.com/google/uuid"
-
 	contribContenttype "github.com/dapr/components-contrib/contenttype"
 	contribPubsub "github.com/dapr/components-contrib/pubsub"
 )
 
 // CloudEvent is a request object to create a Dapr compliant cloudevent.
+// The cloud event properties can manually be overwritten by using metadata beginning with "cloudevent-" as prefix.
 type CloudEvent struct {
-	ID              string
-	Data            []byte
-	Topic           string
-	Pubsub          string
-	DataContentType string
-	TraceID         string
-	TraceState      string
+	ID              string `mapstructure:"id"`
+	Data            []byte `mapstructure:"data"`
+	Topic           string `mapstructure:"topic"`
+	Pubsub          string `mapstructure:"pubsub"`
+	DataContentType string `mapstructure:"datacontenttype"`
+	TraceID         string `mapstructure:"traceid"`
+	TraceState      string `mapstructure:"tracestate"`
+	Source          string `mapstructure:"source"`
+	Type            string `mapstructure:"type"`
+	TraceParent     string `mapstructure:"traceparent"`
+	SpecVersion     string `mapstructure:"specversion"`
+	Time            string `mapstructure:"time"`
 }
 
 // NewCloudEvent encapsulates the creation of a Dapr cloudevent from an existing cloudevent or a raw payload.
 func NewCloudEvent(req *CloudEvent) (map[string]interface{}, error) {
 	if contribContenttype.IsCloudEventContentType(req.DataContentType) {
 		return contribPubsub.FromCloudEvent(req.Data, req.Topic, req.Pubsub, req.TraceID, req.TraceState)
 	}
-	return contribPubsub.NewCloudEventsEnvelope(uuid.New().String(), req.ID, contribPubsub.DefaultCloudEventType,
```

Comment:
FYI the defaults removed here are already handled within the function itself if empty strings are passed, which is exactly what we are doing here assuming the given properties were not overridden.